apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -153,13 +164,13 @@ create(ResId, Group, ResourceType, Config, Opts) ->
     case SpawnBufferWorkers andalso lists:member(QueryMode, [sync, async]) of
         true ->
             %% start resource workers as the query type requires them
-            ok = emqx_resource_buffer_worker_sup:start_workers(ResId, Opts),
-            case maps:get(start_after_created, Opts, ?START_AFTER_CREATED) of
-                true ->
-                    wait_for_ready(ResId, maps:get(start_timeout, Opts, ?START_TIMEOUT));
```

Comment:
I think this wait was inside the `true` clause by pure accident: I can't see a reason why we should wait for the resource to be ready only when spawning buffer workers and if the query mode is `{,a}sync`.  :scream_cat: :see_no_evil:

Reply:
I also don't see any reason so it is probably correct to fix this.